StortSpelProjekt/Engine/src/Renderer/DX12Tasks/QuadTask.cpp

```diff
@@ -32,24 +32,31 @@ QuadTask::~QuadTask()
 
 void QuadTask::SetQuadComponents(std::vector<component::GUI2DComponent*>* quadComponents)
 {
-	m_QuadComponents = *quadComponents;
-
-	// Put each quad in one out of three vectors depending on depth level
-	for (component::GUI2DComponent* qui2DComponent : m_QuadComponents)
+	// If m_Quadcomponents has changed, update the map
+	if (m_QuadComponents != *quadComponents)
```

Comment:
This code is not perforamance crittical, but this != operation compares each element in the array. It would just be faster to clear the quadManagers and then pushback all the components.

Reply:
Okay, I will remove the if-statement